### PR TITLE
Add HumanRobotPosePublisher rpc functionality

### PR DESCRIPTION
**Pull request by Yeshasvitvs**
This PR addresses add rpc functionality to HumanRobotPosePublisher in order to set the robot pose with respect to the human during pHRI experiments. The details related to the modifications done are tracking in https://github.com/robotology/human-dynamics-estimation/issues/104

@lrapetti

### CODE REVIEW

**Comment by Yeshasvitvs** [comment]
@diegoferigo can you please take a quick look at this PR ?
I addressed the changes you mentioned before. 
If there are not major changes needed, I will proceed to merge it to master.

File: publishers/HumanRobotPosePublisher/HumanRobotPosePublisher.cpp [hunk 1]
```diff
@@ -10,36 +10,90 @@
...
     yarp::dev::IFrameTransform* iFrameTransform = nullptr;
 
+    // Final ground to robot base transform
+    yarp::sig::Matrix ground_H_robotBase;
+
     // Human Robot fixed transform
-    iDynTree::Transform humanRobotFixedTransform;
-    yarp::sig::Matrix robotLeftFoot_H_humanLeftFoot;
+    iDynTree::Transform robotHumanFixedTransform;
+    yarp::sig::Matrix humanLeftFoot_H_robotLeftFoot;
```
Comment (diegoferigo) [inline publishers/HumanRobotPosePublisher/HumanRobotPosePublisher.cpp]: Just to double check, is it correct left to left foot? If human and robot are facing, right to left seems more intuitive. I do not remember what we were using in the first place. If this is what you expect, ignore this comment.

File: publishers/HumanStatePublisher/HumanStatePublisher.cpp [hunk 2]
```diff
@@ -257,8 +285,30 @@ void HumanStatePublisher::run()
     // WRITE THE MESSAGES
     // ==================
 
-    pImpl->humanBasePoseROS.publisher.write(/*forceStrict=*/true);
+    //pImpl->humanBasePoseROS.publisher.write(/*forceStrict=*/true);
```
Comment (diegoferigo) [inline publishers/HumanStatePublisher/HumanStatePublisher.cpp]: Leftover?

File: publishers/HumanStatePublisher/HumanStatePublisher.cpp [hunk 3]
```diff
@@ -145,6 +153,26 @@ bool HumanStatePublisher::open(yarp::os::Searchable& config)
...
 
+
+    // =========================
+    // OPEN THE TRANSFORM CLIENT
+    // =========================
+
+    yarp::os::Property options;
+    options.put("device", "transformClient");
+    options.put("local", "/" + DeviceName + "/transformClient");
+    options.put("remote", "/transformServer");
```
Comment (lrapetti) [inline publishers/HumanStatePublisher/HumanStatePublisher.cpp]: The transform client should be opened using the `portprefix` if `hasPortPrefix` is trure (like in https://github.com/robotology/human-dynamics-estimation/blob/devel/publishers/HumanStatePublisher/HumanStatePublisher.cpp#L130)

File: publishers/HumanRobotPosePublisher/HumanRobotPosePublisher.cpp [hunk 4]
```diff
@@ -10,36 +10,90 @@
...
+        }
+        else {
+            this->cmdStatus = false;
+            this->resetStatus = false;
+            return false;
+        }
+
+        yarp::os::ConnectionWriter* reply = connection.getWriter();
+
+        if (reply != NULL) {
```
[the remark this answers is not in the document]
Comment (Yeshasvitvs) [reply publishers/HumanRobotPosePublisher/HumanRobotPosePublisher.cpp]: Done

File: publishers/HumanRobotPosePublisher/HumanRobotPosePublisher.cpp (hunk 1 above)
Comment (Yeshasvitvs) [reply publishers/HumanRobotPosePublisher/HumanRobotPosePublisher.cpp]: > If human and robot are facing, right to left seems more intuitive.

True and that is why i first tried to use the variable name `robotLeftFoot_H_humanLeftFoot` but on checking with the transformation used in HDEv1, I rolled back to using left to left feet transformation as it was double checked correctly.

File: publishers/HumanStatePublisher/HumanStatePublisher.cpp (hunk 2 above)
Comment (Yeshasvitvs) [reply publishers/HumanStatePublisher/HumanStatePublisher.cpp]: Yes

File: publishers/HumanStatePublisher/HumanStatePublisher.cpp (hunk 3 above)
Comment (Yeshasvitvs) [reply publishers/HumanStatePublisher/HumanStatePublisher.cpp]: @lrapetti the changes in this PR are based on the `master` branch. So, for the time being we can merge it to master and while doing an update on the master from `devel` branch the updates will be done correctly.

Comment (lrapetti) [reply publishers/HumanStatePublisher/HumanStatePublisher.cpp]: you are right, we should remember to add it when moving to `devel`

**Comment by Yeshasvitvs** [comment]
Travis fails when looking for [`feature/cleanup` branch](https://github.com/robotology/human-dynamics-estimation/blob/feature/HumanRobotPosePublisher-rpc-functionality/.ci/build_deps.sh#L71) which is deleted on `wearables` repo. @diegoferigo if you agree we can proceed to merging this branch to master.

**Comment by traversaro** [comment]
Why you don't remove the line https://github.com/robotology/human-dynamics-estimation/blob/8cfc88c8fd9fad41ebe837e1c91c59f91df376b9/.ci/build_deps.sh#L71 to fix Travis?

**Comment by Yeshasvitvs** [comment]
> Why you don't remove the line
> 
> https://github.com/robotology/human-dynamics-estimation/blob/8cfc88c8fd9fad41ebe837e1c91c59f91df376b9/.ci/build_deps.sh#L71
> 
> to fix Travis?

You are right  :grin:

**Comment by Yeshasvitvs** [comment]
In the spirit of cleaning up this repo https://github.com/robotology/human-dynamics-estimation/issues/101, I will proceed with merging this branch to the master branch and delete the branch.

CC @lrapetti

**Pull request merged**